### PR TITLE
(RE-4842) Add archival task to nightly_repos.rake

### PR DESCRIPTION
**Pull request by haus**
This commit adds a task to tar up a repo for shipping as part of repo
deployment. This task assumes that prepare_signed_repos has been run
previously for the build.

### CODE REVIEW

File: tasks/nightly_repos.rake [hunk 1]
```diff
@@ -44,6 +44,31 @@ namespace :pl do
...
+          local_target = File.join(Pkg::Config.project, Pkg::Config.ref)
+        elsif versioning == 'version'
+          local_target = File.join(Pkg::Config.project, Pkg::Util::Version.get_dot_version)
+        end
+
+        Dir.chdir(local_target) do
+          if Pkg::Util::File.empty_dir?(path_to_repo)
+            warn "Skipping #{name_of_archive} because it (#{path_to_repo}) has no files"
+          else
+            Pkg::Util::Execution.ex("#{tar} -czf #{File.join("repos", "#{name_of_archive}.tar.gz")} #{path_to_repo}")
```
Comment (mckern) [inline tasks/nightly_repos.rake]: Shelling out to tar, not using `git-archive`?

Comment (haus) [inline tasks/nightly_repos.rake]: git-archive isn't much use here. this isn't a git repo, but a package repo.

Comment (mckern) [inline tasks/nightly_repos.rake]: Got it.

**Comment by shrug** [comment]
:+1:

**Comment by melissa** [comment]
This is so gross... why do we need this?

**Comment by haus** [comment]
@melissa it's in the description of https://tickets.puppetlabs.com/browse/RE-4842

**Comment by melissa** [comment]
The ticket has that information, but the commit message doesn't...

**Comment by haus** [comment]
tl;dr; pe_repo needs to fetch a repo to install puppet-agent from. recursive fetching isn't an option on s3, so a pre-known tarball with the repo for a platform inside is the chosen option.

**Comment by haus** [comment]
@melissa added some extra info to commit message
